### PR TITLE
[fx][pass] Fix type of exception

### PR DESCRIPTION
**Pull request by angelayi**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #84094

@diff-train-skip-merge

### CODE REVIEW

File: torch/fx/passes/infra/pass_manager.py [hunk 1]
```diff
@@ -31,6 +31,8 @@ def wrapped_fn(gm):
         fn(gm)
         return PassResult(gm, True)
 
+    if wrapped_fn.__name__ == 'wrapped_fn':
```
Comment (SherlockNoMad) [inline torch/fx/passes/infra/pass_manager.py:34]: is this really need?IIUC, functools.wraps doesn't change the wrapped function's name.

**Review by SherlockNoMad — APPROVED** [review]
Per comment.

File: torch/fx/passes/infra/pass_manager.py (hunk 1 above)
Comment (angelayi) [reply torch/fx/passes/infra/pass_manager.py:34]: For the passes that are written through PassBase, they don't have a .__name__ field, so the name will end up being "wrapped_fn". This way they will become the name of the object.

**Pull request closed without merging**